Sympy 1.10 removes `sympy.compatibility.core`

This stems from https://github.com/neurostuff/NiMARE/pull/654.

```
ImportError while loading conftest '/home/runner/work/NiMARE/NiMARE/nimare/tests/conftest.py'.
[11](https://github.com/neurostuff/NiMARE/runs/5454997736?check_suite_focus=true#step:5:11)
nimare/__init__.py:11: in <module>
[12](https://github.com/neurostuff/NiMARE/runs/5454997736?check_suite_focus=true#step:5:12)
    from . import (
[13](https://github.com/neurostuff/NiMARE/runs/5454997736?check_suite_focus=true#step:5:13)
nimare/decode/__init__.py:3: in <module>
[14](https://github.com/neurostuff/NiMARE/runs/5454997736?check_suite_focus=true#step:5:14)
    from . import continuous, discrete, encode
[15](https://github.com/neurostuff/NiMARE/runs/5454997736?check_suite_focus=true#step:5:15)
nimare/decode/continuous.py:14: in <module>
[16](https://github.com/neurostuff/NiMARE/runs/5454997736?check_suite_focus=true#step:5:16)
    from ..meta.cbma.base import CBMAEstimator
[17](https://github.com/neurostuff/NiMARE/runs/5454997736?check_suite_focus=true#step:5:17)
nimare/meta/__init__.py:3: in <module>
[18](https://github.com/neurostuff/NiMARE/runs/5454997736?check_suite_focus=true#step:5:18)
    from . import ibma, kernel
[19](https://github.com/neurostuff/NiMARE/runs/5454997736?check_suite_focus=true#step:5:19)
nimare/meta/ibma.py:7: in <module>
[20](https://github.com/neurostuff/NiMARE/runs/5454997736?check_suite_focus=true#step:5:20)
    import pymare
[21](https://github.com/neurostuff/NiMARE/runs/5454997736?check_suite_focus=true#step:5:21)
/opt/hostedtoolcache/Python/3.7.12/x64/lib/python3.7/site-packages/pymare/__init__.py:2: in <module>
[22](https://github.com/neurostuff/NiMARE/runs/5454997736?check_suite_focus=true#step:5:22)
    from .effectsize import (OneSampleEffectSizeConverter,
[23](https://github.com/neurostuff/NiMARE/runs/5454997736?check_suite_focus=true#step:5:23)
/opt/hostedtoolcache/Python/3.7.12/x64/lib/python3.7/site-packages/pymare/effectsize/__init__.py:1: in <module>
[24](https://github.com/neurostuff/NiMARE/runs/5454997736?check_suite_focus=true#step:5:24)
    from .base import (OneSampleEffectSizeConverter, TwoSampleEffectSizeConverter,
[25](https://github.com/neurostuff/NiMARE/runs/5454997736?check_suite_focus=true#step:5:25)
/opt/hostedtoolcache/Python/3.7.12/x64/lib/python3.7/site-packages/pymare/effectsize/base.py:11: in <module>
[26](https://github.com/neurostuff/NiMARE/runs/5454997736?check_suite_focus=true#step:5:26)
    from .expressions import select_expressions
[27](https://github.com/neurostuff/NiMARE/runs/5454997736?check_suite_focus=true#step:5:27)
/opt/hostedtoolcache/Python/3.7.12/x64/lib/python3.7/site-packages/pymare/effectsize/expressions.py:9: in <module>
[28](https://github.com/neurostuff/NiMARE/runs/5454997736?check_suite_focus=true#step:5:28)
    from sympy.core.compatibility import exec_
[29](https://github.com/neurostuff/NiMARE/runs/5454997736?check_suite_focus=true#step:5:29)
E   ImportError: cannot import name 'exec_' from 'sympy.core.compatibility'
```

--- a/pymare/effectsize/expressions.py
+++ b/pymare/effectsize/expressions.py
@@ -5,10 +5,6 @@ from itertools import chain
 from pathlib import Path
 
 from sympy import Symbol, sympify
-from sympy.core.compatibility import exec_
-
-_locals = {}
-exec_("from sympy.stats import *", _locals)
 
 
 class Expression:
@@ -26,9 +22,12 @@ class Expression:
     """
 
     def __init__(self, expression, description=None, type=0):
+        self._locals = {}
+        exec("from sympy.stats import *", self._locals)
+
         self.expr = expression
         self.description = description
-        self.sympy = sympify(expression, locals=_locals)
+        self.sympy = sympify(expression, locals=self._locals)
         self.type = type
         self.symbols = self.sympy.free_symbols
 

